src/Compilers/CSharp/Test/Semantic/FlowAnalysis/LocalFunctions.cs

```diff
@@ -1318,5 +1318,67 @@ void Local()
                 //         Local();
                 Diagnostic(ErrorCode.ERR_UseDefViolationField, "Local()").WithArguments("_x").WithLocation(14, 9));
         }
+
+
+        [Fact]
+        [WorkItem(18813, "https://github.com/dotnet/roslyn/issues/18813")]
+        public void LocalIEnumerableFunctionWithOutParameter1()
+        {
+            var comp = CreateStandardCompilation(@"
+class c
+{
+    static void Main(string[] args)
+    {
+        System.Collections.Generic.IEnumerable<string> getFoo(int count, out bool foobar)
+        {
+            foobar = true;
```

Comment:
Nit: we actually try to avoid "foobar" in the codebase because it triggers some geo-political keyword checker. Call it `output` or something else.